### PR TITLE
Replace SyncManager/AccountManager usage by WorkManager

### LINKED ISSUE #44: Migrate SyncAdapter to WorkManager
Prereq for migrating away from ContentProvider.

### PR DESCRIPTION
**Pull request by strooooke**
Closes #44

### CODE REVIEW

File: app/src/main/java/com/lambdasoup/quickfit/persist/SyncWorker.kt [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    private var contentResolver: ContentResolver = appContext.contentResolver
+    private var googleApiClient: GoogleApiClient? = null
+
+    override fun startWork(): ListenableFuture<Result> =
+            CallbackToFutureAdapter.getFuture { completer ->
+                googleApiClient = GoogleApiClient.Builder(appContext)
+                        .addApi(Fitness.SESSIONS_API)
+                        .addScope(Scope(Scopes.FITNESS_ACTIVITY_READ_WRITE))
+                        .addConnectionCallbacks(
```
Comment (maxhille) [inline app/src/main/java/com/lambdasoup/quickfit/persist/SyncWorker.kt]: too much nesting coming here

File: app/src/main/java/com/lambdasoup/quickfit/persist/SyncWorker.kt [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                                            Timber.d("Found %s sessions to sync", sessionsCursor?.count ?: "no")
+                                            insertNextSession(sessionsCursor, completer, SyncStatus(hasFailedInsertions = false))
+                                        } catch (e: RemoteException) {
+                                            completer.set(Result.retry())
+                                        }
+                                    }
+
+                                    override fun onConnectionSuspended(i: Int) {
+                                        Timber.d("connection suspended")
+                                        if (i == GoogleApiClient.ConnectionCallbacks.CAUSE_NETWORK_LOST) {
```
Comment (maxhille) [inline app/src/main/java/com/lambdasoup/quickfit/persist/SyncWorker.kt]: `i` can probably have a better name

File: app/src/main/java/com/lambdasoup/quickfit/persist/SyncWorker.kt [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        val startTime = cursor.getLong(cursor.getColumnIndex(QuickFitContract.SessionEntry.START_TIME))
+        val endTime = cursor.getLong(cursor.getColumnIndex(QuickFitContract.SessionEntry.END_TIME))
+        val sessionName = cursor.getString(cursor.getColumnIndex(QuickFitContract.SessionEntry.NAME))
+        val sessionBuilder = Session.Builder()
+                .setActivity(cursor.getString(cursor.getColumnIndex(QuickFitContract.SessionEntry.ACTIVITY_TYPE)))
+                .setStartTime(startTime, TimeUnit.MILLISECONDS)
+                .setEndTime(endTime, TimeUnit.MILLISECONDS)
+        if (sessionName != null) {
+            sessionBuilder.setName(sessionName)
+        }
```
Comment (maxhille) [inline app/src/main/java/com/lambdasoup/quickfit/persist/SyncWorker.kt:113]: `else`?

**Review by maxhille — APPROVED** [review]
just some suggestions :+1:

File: app/src/main/java/com/lambdasoup/quickfit/persist/SyncWorker.kt (hunk 2 above)
Comment (strooooke) [reply app/src/main/java/com/lambdasoup/quickfit/persist/SyncWorker.kt]: indeed.

File: app/src/main/java/com/lambdasoup/quickfit/persist/SyncWorker.kt (hunk 3 above)
Comment (strooooke) [reply app/src/main/java/com/lambdasoup/quickfit/persist/SyncWorker.kt:113]: ..else we don't set a name for the session. That's fine.

Could do some abuse of kotlin here, like
```
                .setEndTime(endTime, TimeUnit.MILLISECONDS)
                .apply { 
                    sessionName?.let { setName(it) }
                }
```

but I see no advantage there. In fact, this relies on the fact that the Builder will mutate _and_ return itself.

Comment (maxhille) [reply app/src/main/java/com/lambdasoup/quickfit/persist/SyncWorker.kt:113]: I did not know that not having a name was ok. Just made sure you give this a second thought. Maybe `#setName(String)` can also be called with `null`, then you would not need the `if` in the first place.

**Pull request merged**